brokers/che-plugin-broker/broker/broker_test.go

```diff
@@ -0,0 +1,420 @@
+//
+// Copyright (c) 2012-2018 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package broker
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	yaml "gopkg.in/yaml.v2"
+
+	tests "github.com/eclipse/che-plugin-broker/brokers_test"
+	cmock "github.com/eclipse/che-plugin-broker/common/mocks"
+	fmock "github.com/eclipse/che-plugin-broker/files/mocks"
+	"github.com/eclipse/che-plugin-broker/model"
+	"github.com/eclipse/che-plugin-broker/storage"
+)
+
+var (
+	broker     = NewBroker()
+	bMock      = &cmock.Broker{}
+	uMock      = &fmock.IoUtil{}
+	mockBroker = &ChePluginBroker{
```

Comment:
@garagatyi You created real object Broker here. Why it's called mockBroker, maybe "broker" is simplier?

Reply:
There are 2 brokers for different cases:
1. a real one which is called `broker`
1. `mockBroker` which uses mocked fields and allows to setup its behavior with stubbing.